services:
  postgres:
    container_name: forgejo-postgres
    image: postgres:17.6-alpine@sha256:c5da6bc9da9e3a3b6dc85585078c05c002c2d4deeaca00479f5571e2d00ca5d2
    restart: unless-stopped
    environment:
      - TZ=UTC
    env_file:
      - /custom/files/forgejo.env
    networks:
      - forgejo
    volumes:
      - 'postgres_data:/var/lib/postgresql/data'
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "forgejo"]
      start_period: 10s
      interval: 10s
      timeout: 5s
      retries: 3
  forgejo:
    container_name: forgejo
    image: codeberg.org/forgejo/forgejo:13.0.1-rootless@sha256:fe7e2562e1d499dca8b58ba2d91c38f5354b8237eca6a82be32c4cadcb183786
    restart: unless-stopped
    environment:
      - TZ=America/New_York
    env_file:
      - /custom/files/forgejo.env
    networks:
      - forgejo
      - proxy
    depends_on:
      - postgres
    volumes:
      - 'forgejo_config:/etc/gitea'
      - 'forgejo_data:/var/lib/gitea'
    labels:
      - traefik.enable=true
      - traefik.http.routers.forgejo.rule=Host(`forgejo.${SECRET_INTERNAL_DOMAIN_NAME}`)
      - traefik.http.routers.forgejo.tls=true
      - traefik.http.routers.forgejo.tls.certresolver=cloudflare
      - traefik.http.routers.forgejo.entrypoints=websecure
      - traefik.http.services.forgejo.loadbalancer.server.port=3000

networks:
  forgejo:
  proxy:
    external: true

volumes:
  forgejo_config:
    driver: local
  forgejo_data:
    driver: local
  postgres_data:
    driver: local
